viite-UI/src/view/SelectToolControl.js

```diff
@@ -0,0 +1,230 @@
+            condition: function(event){ return ol.events.condition.platformModifierKeyOnly(event); }
+        });
+
+        var drawInteraction = new ol.interaction.Draw({
+            condition: function(event){ return ol.events.condition.noModifierKeys(event) || isPolygonActive; },
+            type: ('Polygon'),
+            style: drawStyle()
+        });
+
+        var drawSquare = new ol.interaction.Draw({
+            condition: function(event){ return ol.events.condition.noModifierKeys(event) || isRectangleActive; },
+            type: ('Circle'),
+            style: drawStyle(),
+            geometryFunction: ol.interaction.Draw.createBox()
+        });
+
+        var selectInteraction = new ol.interaction.Select({
+            layer: layer,
+            condition: function(events){
+                return !isPolygonActive && !isRectangleActive && enabled &&(ol.events.condition.doubleClick(events) || ol.events.condition.singleClick(events));
+            },
+            style: settings.style,
+            filter : settings.filterGeometry
+        });
+
+        selectInteraction.set('name', layer.get('name'));
+
+        dragBoxInteraction.on('boxstart', settings.onDragStart);
+        dragBoxInteraction.on('boxend', function() {
+            var extent = dragBoxInteraction.getGeometry().getExtent();
+            interactionEnd(extent);
+        });
+
+        function interactionEnd(extent) {
+            var selectedFeatures = [];
+            layer.getSource().forEachFeatureIntersectingExtent(extent, function (feature) {
+                selectedFeatures.push(feature.getProperties());
+            });
+            settings.onInteractionEnd(selectedFeatures);
+        }
+
+        drawSquare.on('drawend', function(evt) {
+            evt.preventDefault();
+            var extent = evt.feature.getGeometry().getExtent();
+            interactionEnd(extent);
+        });
+
+        drawInteraction.on('drawend', function(evt){
+            evt.preventDefault();
+            var polygonGeometry = evt.feature.getGeometry();
+            var features =  layer.getSource().getFeatures();
+            var selected = _.filter(features, function(feature) {
+                return _.some(feature.getGeometry().getCoordinates(), function(coordinate) {
+                    return polygonGeometry.intersectsCoordinate(coordinate);
+                });
+              });
+            var selectedProperties = _.map(selected, function(select) { return select.getProperties(); });
+            settings.onInteractionEnd(selectedProperties);
+        });
+
+        selectInteraction.on('select',  function(evt){
+            if(evt.selected.length > 0 && settings.enableSelect(evt))
+                unhighlightLayer();
+            else
+                highlightLayer();
+
+            settings.onSelect(evt);
+        });
+
+        var toggleDragBox = function() {
+          // if (!application.isReadOnly() && enabled && settings.draggable) {
```

Comment:
Is this part needed or can it be safely removed?

Reply:
yep needs cleanup